compact_str/src/repr/arc/mod.rs

```diff
@@ -6,12 +6,15 @@ use std::sync::atomic::Ordering;
 use std::{
```

Comment:
this module (`arc`) isn't actually used any more, all of the code is dead. I kept it around to possibly use in the future, but it really should get deleted.

I'll make a change to delete the module, or you can go ahead and delete it too. This should remove the need for the `addr_of_mut` API, which should fix the MSRV issue!

Reply:
I'll delete it.

Reply:
Thanks!